### ISSUE TITLE
Microsoft Ajax Content Delivery Network

### ISSUE THREAD

**Issue by Rick-Anderson** [aspnet/Docs#5832]
Use this issue to report problems with the [Microsoft Ajax Content Delivery Network](https://docs.microsoft.com/en-us/aspnet/ajax/cdn/overview)

**Comment by denisyfrolov** [comment]
Looks like the CDN is pretty dead currently :-(

**Comment by ChrisSfanos** [comment]
Hey @denisyfrolov - I just hit https://docs.microsoft.com/en-us/aspnet/ajax/cdn/overview and also tried to download a jQuery file and both worked.  Are you seeing something different?

**Comment by denisyfrolov** [comment]
@ChrisSfanos I am. ERR_CONNECTION_TIMED_OUT from my location (Russia). I'm receiving cs9.wpc.v0cdn.net host

**Comment by denisyfrolov** [comment]
Sh.., Seems like the IP address was blocked here in Russia by government communication department http://blocklist.rkn.gov.ru/. I have no idea why they did it, but pretty similar they were blocking someone else and puted the address by mistake like just why not. Sort of... 
![sc1](https://user-images.githubusercontent.com/5620027/38837196-ba006008-41d9-11e8-873b-dacb6403f67d.png)
![sc2](https://user-images.githubusercontent.com/5620027/38837203-be7bbc18-41d9-11e8-9cc1-93bb21e17b74.png)

**Comment by blowdart** [comment]
Oh dear. This is a CDN so it hosts more things than just the asp.net cdn, heck it could be part of the attempts to block Telegraph, I doubt there's anything we can do here unfortunately :(

**Comment by denisyfrolov** [comment]
@blowdart sounds catastrophically. Do you understand that the CDN is used by giant numbers of sites including Russian ones. And is uses for many of Microsoft's sites including Azure Portal for instance.

**Comment by blowdart** [comment]
I do, but Russia controls their bloke list here. We will talk to the CDN hosts, but if they are being blocked for other things they're caching then it's just going to be blocked again.

@ChrisSfanos can you open a support ticket please?

**Comment by denisyfrolov** [comment]
@blowdart I do understand you can't change the blocking list, but c'mon, you have to make something up with that, atleast a mirror or something like that. Otherwise there will be really huge problem to many of people here in Russia. The RKN's blocking is pretty slow process for many of local ISPs but it will be global sooner or later.

**Comment by blowdart** [comment]
The CDN *is* the mirror. That's the point of it unfortunately. Spinning up a new site would mean you either change code to use a new host, or we redirect everyone to the mirror, which isn't sustainable in terms of traffic, which is why we offload to a CDN in the first place.

**Comment by Rick-Anderson** [comment]
@denisyfrolov  per the [CDN](https://docs.microsoft.com/en-us/aspnet/ajax/cdn/overview) article:

>Production applications should not take a hard dependency on CDN assets. Applications should test for the CDN asset referenced, and use a fallback asset when the CDN is not available.

**Comment by denisyfrolov** [comment]
@Rick-Anderson yeah, damn good idea. Who will tell this to the Azure Portal developers?
![sc3](https://user-images.githubusercontent.com/5620027/38840468-7e3200f4-41e8-11e8-8a57-92317a1ae04d.png)

**Comment by denisyfrolov** [comment]
@Rick-Anderson and don't forget about MSDN's blog developers please..

![sc5](https://user-images.githubusercontent.com/5620027/38841461-2fe636d6-41ed-11e8-828c-e3317ca811f0.png)

**Comment by denisyfrolov** [comment]
@Rick-Anderson shall I continue?

**Comment by otryshko** [comment]
@denisyfrolov what if tomorrow *.microsoft.com will be blocked (because why not?) More practical approach would be to use a VPN like Chinese do, no?

**Comment by denisyfrolov** [comment]
@otryshko well, probably it sounds like a plan.. in long term.. but how it is supposed to help to consumers now?) let's solve problems as they occur

**Comment by Rick-Anderson** [comment]
@denisyfrolov Thanks for the report. I opened a bug with the Azure folks.

**Comment by denisyfrolov** [comment]
@Rick-Anderson That's not about the Azure's sites only. Almost each of Microsoft's sites is absolutely unusable in Russia now, like: Docs, MSDN, Visual Studio, Office, Windows, Xbox, all of them and many others are almost dead now.

**Comment by Rick-Anderson** [comment]
@denisyfrolov I'm opening up bugs.

**Comment by davidebbo** [comment]
@denisyfrolov are you sure that it really affects everyone in Russia? I would think that this would make big news, and I'm not finding any mentions of it in the news or on Twitter.

**Comment by denisyfrolov** [comment]
@davidebbo oh yes I am, it started at night.. Here is 5 am now in Moscow, people sleep yet. Also It depends on ISP, many of them do the blocking very slow. But from all of the country I'm receiving reports that Microsoft's sites are dying. It is not about Microsoft only, many of IP addresses of AWS are also blocked. .

**Comment by denisyfrolov** [comment]
@davidebbo here is the latest news by this topic https://www.bleepingcomputer.com/news/government/russia-bans-18-million-amazon-and-google-ips-in-attempt-to-block-telegram/

**Comment by pronichkin** [comment]
@denisyfrolov "Almost each of Microsoft's sites is absolutely unusable in Russia now, like: Docs, MSDN, Visual Studio, Office, Windows, Xbox, all of them and many others are almost dead now."

For good or bad, I cannot confirm that, at least right now. (11 AM Moscow time.) All of the sites you mentioned are currently working for me. They are a bit slower then usual, but open successfully without noticeable errors. (Not using any VPN here of course.)

**Comment by tr3x33** [comment]
@denisyfrolov 

I was not able to reproduce the mentioned issue.

> Docs, MSDN, Visual Studio, Office, Windows, Xbox

I checked every from this list and all of them fine, moreover, I used Azure Portal earlier today without any issues. I'm located in Russia as well.

So is there a chance to reproduce it somehow?

**Comment by pronichkin** [comment]
@phpony

indeed, this particular URL does not work for me either.

**Comment by phpony** [comment]
For example: https://docs.microsoft.com/en-us/aspnet/ajax/cdn/overview page uses jQuery from Microsoft Ajax Content Delivery Network, so it loads, but very slow and without any js functionality:

![роскомпозор](https://i.imgur.com/NLIfAzC.png)

**Comment by ChALkeR** [comment]
web.skype.com is also effectively blocked (html loads, js doesn't).

**Comment by tr3x33** [comment]
@phpony ye, thanks, that's particular URL for jquery-3.3.1.js doesn't work for me as well, but still, I don't see how it affects the mentioned MS websites

**Comment by phpony** [comment]
@tr3x33 well, you know, Microsoft for some unknown reason is using their own CDN on their sites. And as for now when CDN is blocked those sites are not working as intended. And some of them are **completely broken**:

![web.skype.com](https://i.imgur.com/j65Zmts.png)

**Comment by ChALkeR** [comment]
https://web.skype.com:
* https://s4w.cdn.skype.com/0-239-0/js/checkBrowserSupport.js
* https://s4w.cdn.skype.com/0-239-0/js/index.js

https://azure.microsoft.com:
* https://amp.azure.net/libs/amp/1.7.4/skins/amp-default/azuremediaplayer.min.css
* https://azurecomcdn.azureedge.net/cvt-267051159310fb9e9ab3701a1c39e064f60b7457875daefb451f39f145d7d8ae/bundles/home.css
* https://azurecomcdn.azureedge.net/cvt-de4b3c3d1dc2506b6693f0f98884e1dc074cda9d66cab39b7b48a115fdfc4c0f/scripts/sundog/lib/jquery-2.1.4.min.js
* … some more from the same domains …

**Comment by Pzixel** [comment]
This is an official governement site where you can check if site is blocked: https://blocklist.rkn.gov.ru

And it clearly says that at least some addresses are blocked:
![image](https://user-images.githubusercontent.com/11201122/38862094-ba52b872-423c-11e8-9467-c1c05e0d3731.png)

Use google translait for the page if neccessary.

**Comment by lemenkov** [comment]
This should be closed as "CLOSED, CHANGEYOURGOVERNMENT"

**Comment by visual4** [comment]
currently not reachable from Germany too?
I get timeouts for https://ajax.aspnetcdn.com/ajax/3.5/MicrosoftAjax.js and jQuery within MS-Sites

Cheers,
Björn

**Comment by ChrisSfanos** [comment]
We are hearing reports of this from Europe.  We've asked Azure/Verizon to investigate

**Comment by Mekanik** [comment]
Azure App-insight availability test start failing from all Europe regions because of CDN `Unable to connect to the remote server`. Started approximately 2 hours ago.
![image](https://user-images.githubusercontent.com/670361/38893698-e653788c-423f-11e8-9edb-af2d67e81f02.png)

**Comment by phpony** [comment]
To the good news - CDN is now resolving to 93.184.221.201 instead of 93.184.221.200 and is available for me.

**Comment by Vertigo093i** [comment]
93.184.221.201 is already blocked too. And even next IP address 152.199.20.1 is blocked as of today.
https://reestr.rublacklist.net/rec/349558/

**Comment by denisyfrolov** [comment]
The CDN is blocked again.

**Comment by ChALkeR** [comment]
https://github.com/ChALkeR/fuckrkn-dnsmasq/blob/master/fuckrkn.conf — this kinda fixes it for me (for now).

**Comment by hjalalisharepoint2013** [comment]
Please consider updating `https://ajax.aspnetcdn.com/ajax/mvc/3.0/jquery.unobtrusive-ajax.js`, Its still looking for `live` function in jquery!
`Uncaught TypeError: a(...).live is not a function
    at jquery.unobtrusive-ajax.min.js:5`

**Comment by Pzixel** [comment]
@hjalalisharepoint2013 yep, it's definitely looks like a bug, this page is trying to download jquery in year 2018.

**Comment by JustCallMeAD** [comment]
Unfortunately I tried emailing a request to "AjaxCDNSubmission@microsoft.com" but my message did not go though.. Not sure why it is advertised on a public page if it does not work - https://docs.microsoft.com/en-us/aspnet/ajax/cdn/overview

_Your message to AjaxCDNSubmission@microsoft.com couldn't be delivered.
The group AjaxCDNSubmission only accepts messages from people in its organization or on its allowed senders list, and your email address isn't on the list._

**Can someone please allow "AjaxCDNSubmission@microsoft.com" to accept emails from the public & also please look into my request below. Thanks :)**

Please add another CDN link for Bootstrap 4 (bootstrap.bundle.js).

To ensure that the new asp.net templates are able to use Bootstrap 4 properly can you please add another CDN link to "bootstrap.bundle.js" (this includes both bootstrap.js & popper.js in 1 js file). 

Please note that popper.js it is now a requirement of Bootstrap 4 & is used in Dropdowns & Tooltips.

See the links below for more information:
https://www.bootstrapcdn.com/
https://getbootstrap.com/docs/4.1/getting-started/introduction/#js
https://getbootstrap.com/docs/4.1/getting-started/download/#bootstrapcdn
https://getbootstrap.com/docs/4.1/getting-started/contents/

Please also see github issues below: 
https://github.com/aspnet/Home/issues/3290
https://github.com/aspnet/Templating/issues/601

**Comment by klabranche** [comment]
Bootstrap 4.1.1 has a xss exploit in it.  https://snyk.io/vuln/npm:bootstrap:20180529  Any idea of when 4.1.2 or 4.1.3 will be available in the CDN?

**Comment by ChALkeR** [comment]
@klabranche that's not an exploit, it's a vulnerability.

**Comment by klabranche** [comment]
@ChALkeR - yes, slip of the keyboard. :-)  updated my comment to reflect.

**Comment by vanillajonathan** [comment]
It would be good if subresource integrity (SRI) hashes for each file were listed on the webpage.

**Comment by nj** [comment]
@hjalalisharepoint2013 Hope you have switched to [newer MVC version, either v4 or v5](https://docs.microsoft.com/en-us/aspnet/ajax/cdn/overview#ASPNET_MVC_Releases_on_the_CDN_16), instead of running that rather old version :wink:

@klabranche You could use the one from Bootstrap in the meantime, so you get it directly. But yeah, oddly enought not updated yet though 2 months have passed.

**Comment by guardrex** [comment]
@footed reports ...

> I use .net framework to build a webform website with ScriptManager and I set EnableCdn to true.
I find ScriptManager set the MicrosoftAjax.js to https://ajax.aspnetcdn.com/ajax/4.6/1/MicrosoftAjax.zh-Hant.js and the response code is 404 so the system come back to my website to get the MicrosoftAjax.zh-Hant.js and MicrosoftAjaxWebForms.zh-Hant.js
I also try to link MicrosoftAjax.js and MicrosoftAjaxWebForms.zh-Hant.js use another language like fr-Fr the response code is 404 same with zh-Hant.

@ChrisSfanos

**Comment by TheDestinyGroup** [comment]
Hi,

Can you advise when you might add add jquery.validate 1.19.0 and bootstrap 3.4 to the CDN?

Les

**Comment by ChrisSfanos** [comment]
I'm working to publish Validate 1.19 now.  I don't plan to publish Bootstrap 3.4 but will work on 4.2.1 tomorrow

**Comment by TheDestinyGroup** [comment]
Chris,

Thanks for that.

Can you elaborate on why you won't be publishing Bootstrap 3.4 as it contains crucial security updates for Bootstrap 3.x users who are yet to update their sites to support Bootstrap 4.x?

I would appreciate if you could also add jquery.validate.unobtrusive 3.2.11 as the 3.2.x release is referenced by .net core project templates.

I updated my projects to use the cloudflare cdn links for now as they seem to keep up to date with new releases, but I would prefer to use the Ajax CDN links.

Les

**Comment by ChrisSfanos** [comment]
Bootstrap 3.4.0 and 4.2.1 are now live
So I'm checking on .unobtrusive 3.2.11 and I only see the main file on GitHub and not the .min version?

**Comment by TheDestinyGroup** [comment]
Chris,

Thanks for that.

I'm not sure where the other CDN providers got their version from.

The NuGet 3.2.11 package Microsoft provide for .Net 4.x projects doesn't include a minified version.

Les

**Comment by footed** [comment]
> @footed reports (from #10133) ...
> 
> > I use .net framework to build a webform website with ScriptManager and I set EnableCdn to true.
> > I find ScriptManager set the MicrosoftAjax.js to https://ajax.aspnetcdn.com/ajax/4.6/1/MicrosoftAjax.zh-Hant.js and the response code is 404 so the system come back to my website to get the MicrosoftAjax.zh-Hant.js and MicrosoftAjaxWebForms.zh-Hant.js
> > I also try to link MicrosoftAjax.js and MicrosoftAjaxWebForms.zh-Hant.js use another language like fr-Fr the response code is 404 same with zh-Hant.
> 
> @ChrisSfanos

Did somebody try to solve this issue?

**Comment by ChrisSfanos** [comment]
Have we previously published language-specific versions of those files previously?
